yb-voyager/cmd/callhome_payload_builder.go

```diff
@@ -413,3 +455,102 @@ func getAnonymizedIndexObjectsFromIndexToTableMap(indexToTableMap map[string][]s
 	}
 	return objects
 }
+
+// ============================compare performance callhome payload information============================
+
+func packAndSendComparePerformancePayload(status string, errorMsg error, comparator *compareperf.QueryPerformanceComparator) {
+	if !shouldSendCallhome() {
+		return
+	}
+
+	payload := createCallhomePayload()
+	payload.MigrationPhase = COMPARE_PERFORMANCE_PHASE
+	payload.Status = status
+
+	comparePerformancePayload := buildCallhomeComparePerformancePayload(comparator, errorMsg)
+	payload.PhasePayload = callhome.MarshalledJsonString(comparePerformancePayload)
+	err := callhome.SendPayload(&payload)
+	if err == nil && (status == COMPLETE || status == ERROR) {
+		callHomeErrorOrCompletePayloadSent = true
+	}
+}
+
+// buildCallhomeComparePerformancePayload builds the payload for compare performance callhome
+// Collects performance comparison metrics from the comparator for matched queries only
+func buildCallhomeComparePerformancePayload(comparator *compareperf.QueryPerformanceComparator, errorMsg error) *callhome.ComparePerformancePhasePayload {
+	payload := callhome.ComparePerformancePhasePayload{
+		PayloadVersion: callhome.COMPARE_PERFORMANCE_PAYLOAD_VERSION,
+		Error:          callhome.SanitizeErrorMsg(errorMsg, anonymizer),
+	}
+
+	if comparator == nil || comparator.Report == nil {
+		return &payload
+	}
+
+	var queryMetrics []callhome.QueryMetric
+	for _, comparison := range comparator.Report.AllComparisons {
+		if comparison.MatchStatus != compareperf.MATCHED {
+			continue
+		}
+
+		queryMetric := callhome.QueryMetric{
+			QueryLabel:    generateQueryLabel(comparison.Query),
+			SlowdownRatio: comparison.SlowdownRatio,
+			ImpactScore:   comparison.ImpactScore,
+			SourceStats: callhome.QueryStatsCallhome{
+				ExecutionCount:  comparison.SourceStats.ExecutionCount,
+				RowsProcessed:   comparison.SourceStats.RowsProcessed,
+				TotalExecTime:   comparison.SourceStats.TotalExecTime,
+				AverageExecTime: comparison.SourceStats.AverageExecTime,
+				MinExecTime:     comparison.SourceStats.MinExecTime,
+				MaxExecTime:     comparison.SourceStats.MaxExecTime,
+			},
+			TargetStats: callhome.QueryStatsCallhome{
+				ExecutionCount:  comparison.TargetStats.ExecutionCount,
+				RowsProcessed:   comparison.TargetStats.RowsProcessed,
+				TotalExecTime:   comparison.TargetStats.TotalExecTime,
+				AverageExecTime: comparison.TargetStats.AverageExecTime,
+				MinExecTime:     comparison.TargetStats.MinExecTime,
+				MaxExecTime:     comparison.TargetStats.MaxExecTime,
+			},
+		}
+		queryMetrics = append(queryMetrics, queryMetric)
+	}
+
+	// Populate summary and metrics
+	payload.TotalQueries = comparator.Report.Summary.TotalQueries
+	payload.MatchedQueries = comparator.Report.Summary.MatchedQueries
+	payload.SourceOnlyQueries = comparator.Report.Summary.SourceOnlyQueries
+	payload.TargetOnlyQueries = comparator.Report.Summary.TargetOnlyQueries
```

Comment:
similarly here handle `Report.Summary` nil

Reply:
summary field is not a pointer. in worse case the values will be empty